### PR TITLE
Aqcu 1785

### PR DESCRIPTION
**Pull request by kmschoep-usgs**

### CODE REVIEW

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java [hunk 1]
```diff
@@ -11,51 +11,59 @@
...
+	                } else if(pointValue.compareTo(minValue) == 0) {
+	                    minSet.add(point);
+	                }
+	            }
+	        }
+	
+	        result.setMaxPoints(maxSet);
+	        result.setMinPoints(minSet);
+        } catch (Exception e) {
+        	log.error("Exception in findMinMaxPoints: ", e.getMessage());
```
Comment (zmoore-usgs) [inline src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: Should we re-throw the error here or is it okay for this method to return a blank result when an error occurs?

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java [hunk 2]
```diff
@@ -66,28 +74,36 @@ public TimeSeriesMinMax findMinMaxMatchingPoints(TimeSeriesMinMax primaryMinMax,
...
+	
+	            for(TimeSeriesPoint primaryPoint : primaryPoints) {
+	                TimeSeriesPoint relatedPoint = relatedPointMap.get(primaryPoint.getTimestamp().DateTimeOffset);
+	                if(relatedPoint != null) {
+	                    matchingPoints.add(relatedPoint);
+	                }
+	            }
+	        }
+        } catch (Exception e) {
+        	log.error("Exception in findMatchingPoints: ", e.getMessage());
```
Comment (zmoore-usgs) [inline src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: [See above]

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java [hunk 3]
```diff
@@ -66,28 +74,36 @@ public TimeSeriesMinMax findMinMaxMatchingPoints(TimeSeriesMinMax primaryMinMax,
...
 
     protected Map<Instant, TimeSeriesPoint> pointListToMap(List<TimeSeriesPoint> pointList) {
-        if(pointList != null && !pointList.isEmpty()) {
-            return pointList.stream().collect(Collectors.toMap(t -> t.getTimestamp().getDateTimeOffset(), t->t));
+    	try {
+	        if(pointList != null && !pointList.isEmpty()) {
+	            return pointList.stream().collect(Collectors.toMap(t -> t.getTimestamp().getDateTimeOffset(), t->t));
+	        } 
+    	} catch (Exception e) {
+        	log.error("Exception in pointListToMap: ", e.getMessage());
```
Comment (zmoore-usgs) [inline src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: [See above]

File: src/main/java/gov/usgs/aqcu/builder/ReportBuilderService.java [hunk 4]
```diff
@@ -46,108 +58,117 @@ public ReportBuilderService(
...
+	
+			//Report Metadata
+			report.setReportMetadata(getReportMetadata(requestParameters,
+				timeSeriesDescriptions,
+				primaryDescription,
+				requestingUser,
+				qualifiers
+			));
+		} catch (Exception e) {
+			log.error("Exception in buildReport: ", e.getMessage());
```
Comment (zmoore-usgs) [inline src/main/java/gov/usgs/aqcu/builder/ReportBuilderService.java]: [See above]

File: src/main/java/gov/usgs/aqcu/builder/ReportBuilderService.java [hunk 5]
```diff
@@ -157,54 +178,57 @@ public ExtremesReport buildReport(ExtremesRequestParameters requestParameters, S
...
+						timeSeriesDescriptions.get(requestParameters.getUpchainTimeseriesIdentifier()).getParameter());
+				metadata.setUpchainUnit(
+						timeSeriesDescriptions.get(requestParameters.getUpchainTimeseriesIdentifier()).getUnit());
+			}
+			
+			if(qualifierList != null && !qualifierList.isEmpty()) {
+				metadata.setQualifierMetadata(qualifierLookupService.getByQualifierList(qualifierList));
+			}
+		} catch (Exception e) {
+			log.error("Exception in getReportMetadata: ", e.getMessage());
```
Comment (zmoore-usgs) [inline src/main/java/gov/usgs/aqcu/builder/ReportBuilderService.java]: [See above]

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java (hunk 1 above)
Comment (kmschoep-usgs) [reply src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: It's ok for nothing to be returned.

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java (hunk 2 above)
Comment (kmschoep-usgs) [reply src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: It's ok for nothing to be returned.

File: src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java (hunk 3 above)
Comment (kmschoep-usgs) [reply src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java]: It's ok for nothing to be returned.

**Pull request merged**